### PR TITLE
More (minor) plot types gallery fixes.

### PR DESCRIPTION
**Pull request by timhoffm**
## PR Summary

All changes are small. I thought it's ok to put them in a single PR to not run CI that often.

- Include new plots into gallery ordering
- Use a second style for plot types gallery entries without grid - to declutter the actual plotting code more (removes `ax.grid(False)`
- Minor formatting fixes
- Simplify unstructured data generation: There is no need to generate a regular grid and statistically choose points from that. Instead, we can immediately choose random points.

### CODE REVIEW

**Review by tacaswell — APPROVED** [review]

**Comment by tacaswell** [comment]
There is no problem renaming the style as it has not been released yet.

**Comment by timhoffm** [comment]
Maybe we should claim the style for internal use only and name it `_mpl-gallery`?

**Comment by tacaswell** [comment]
I can see arguments both ways.  We use it as part of the user-facing docs so we should be able to change it at will so it should be private, but people may want to mimic and track the style and hence it should be public.  Maybe a note someplace that we consider this style volatile?

**Comment by timhoffm** [comment]
> I can see arguments both ways. We use it as part of the user-facing docs so we should be able to change it at will so it should be private, but people may want to mimic and track the style and hence it should be public.

Both are not possible at the same time, in particular if change includes rename or delete. I don't see why it is critical for anybody to exactly track the style. It's not the end of the world if somebody copies it and doesn't immediately realize when we change. If they still want to track, I think setting up a CI up-to-date check on their side is the correct choice.
For now, I'll be defensive and claim the style to be explicitly private. We can always publicize it later when a compelling reason arises.

File: lib/matplotlib/mpl-data/stylelib/_mpl-gallery.mplstyle [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# from the Matplotlib cheatsheet as used in our gallery:
+# This style is used for the plot_types gallery. It is considered private.
```
Comment (story645) [inline lib/matplotlib/mpl-data/stylelib/_mpl-gallery.mplstyle:1]:
```suggestion
# This style is used for the plot_types gallery. It is considered part of the private API.
```
explicit better than implicit?

**Review by story645 — APPROVED** [review]

File: lib/matplotlib/mpl-data/stylelib/_mpl-gallery.mplstyle (hunk 1 above)
Comment (timhoffm) [reply lib/matplotlib/mpl-data/stylelib/_mpl-gallery.mplstyle:1]: Sure.

**Comment by jklymak** [comment]
Anyone should merge once the build is done.

**Pull request merged**